### PR TITLE
etcd healthz, hotloop fixes

### PR DESCRIPTION
**Pull request by liggitt**
* fixes CPU hotloop in etcd client (pending upstream review, needs picking to our branch and update-deps)
* fixes etcd healthz check to use an actual client (makes it less vulnerable to spurious dns lookup timeouts)
* fixes typo in healthz filter exemption

### CODE REVIEW

**Comment by liggitt** [comment]
@mfojtik @deads2k please review, then I'll move the etcd client commit to our repo branch

File: pkg/cmd/server/origin/webconsole_proxy.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func withAssetServerRedirect(handler http.Handler, accessor *webConsolePublicURL
 func (c *MasterConfig) withConsoleRedirection(handler, assetServerHandler http.Handler, accessor *webConsolePublicURLAccessor) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		// blacklist well known paths so we do not risk recursion deadlocks
-		for _, prefix := range []string{"/apis", "/api", "/oapi", "/healtz", "/version"} {
```
Comment (deads2k) [inline pkg/cmd/server/origin/webconsole_proxy.go]: @simo5 awkward. :)

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/options/etcd.go [hunk 2]
```diff
@@ -166,29 +166,30 @@ func (s *EtcdOptions) ApplyTo(c *server.Config) error {
...
+	if err := s.addEtcdHealthEndpoint(c); err != nil {
+		return err
+	}
 	c.RESTOptionsGetter = &storageFactoryRestOptionsFactory{Options: *s, StorageFactory: factory}
 	return nil
 }
 
-func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) {
+func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
+	healthCheck, err := storagefactory.CreateHealthCheck(s.StorageConfig)
```
Comment (deads2k) [inline vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/options/etcd.go]: pre-existing, but if I called `ApplyWithStorageFactoryTo` wouldn't I expect to use the StorageConfig that's available from the StorageFactory?

Comment (liggitt) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/options/etcd.go]: yes, though we lack the information required to get the specific config to set up the health check here. I don't see that getting addressed as part of this

File: pkg/cmd/server/origin/webconsole_proxy.go (hunk 1 above)
Comment (simo5) [reply pkg/cmd/server/origin/webconsole_proxy.go]: Uups, I guess healthz was not the one that cause the loop :-)

**Comment by deads2k** [comment]
lgtm

**Comment by runcom** [comment]
fix for cadvisor went in (sorry for that.)

**Comment by liggitt** [comment]
tagging per https://github.com/openshift/origin/pull/19992#issuecomment-397013038

**Pull request merged**